### PR TITLE
Added draft for migration guidelines

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added draft for migration guidelines

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

### CODE REVIEW

**Comment by MariusStorhaug** [comment]
I would call it upgrade guidelines but minor thing really.

**Comment by MrMCake** [comment]
> I would call it upgrade guidelines but minor thing really.

done

File: docs/wiki/Upgrade from 0.3.1 to 0.4.0.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Migrating to this change (by overwriting the original module templates/outputs) won't affect the CI pipelines and you can use the environment as is. However, if you built solutions using relative paths to the modules, make sure you update any output reference of the same.
+
+## Parameter file name prefix
+When migrating, you may notice that many of the resource names in the module parameter files have a placeholder `<<namePrefix>>`. This placeholder is automatically replaced when executing any of the CI pipelines and as such requires changes to the [CI environment](#nameprefix-in-settingsjson) too.
+
+You can remove these placeholders as you see fit, or migrate them together with the changes explained in the [`General pipeline updates`](#general-pipeline-updates) & [` NamePrefix in settings.json`](#nameprefix-in-settingsjson) sections of this guide.
+
+## Version file
+A `version.json` file was added to each individual module. Please make sure such a file is added to every module folder in the `arm` folder structure.
+
```
Comment (simonbMS) [inline docs/wiki/Upgrade from 0.3.1 to 0.4.0.md:45]:
```suggestion

For more information on how versioning is handled see [here](https://github.com/Azure/ResourceModules/wiki/PipelinesDesign#publish).

```

File: docs/wiki/Upgrade from 0.3.1 to 0.4.0.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+General updates can be migrated by essentially copying/replacing any code outside of the `arm` folder (i.e. `.github`/`.azuredevops`/`utilities`/`docs`). If you modified the CI environment in the meantime, make sure to carefully merge any conflicting code. Noteworthy changes include:
+-	renamed composite actions / templates,
+-	extended placeholder handling,
+-	extended pipeline functionality (e.g., managementgroup-level removal),
+-	publishing of child modules and
+-	overall extended publishing using `version.json` files
+-	additional deployments in the dependency pipeline
+
+## NamePrefix in settings.json
+A change that affects all modules is the introduction of the `namePrefix` property in the local `settings.json` file. This prefix is used to replace any placeholder such as `<<namePrefix>>` in any parameter file deployed in the CI pipelines.
```
Comment (simonbMS) [inline docs/wiki/Upgrade from 0.3.1 to 0.4.0.md:60]:
```suggestion
A change that affects all modules is the introduction of the `namePrefix` property in the local `settings.json` file.
This prefix is used to simplify the introduction of a "personalized" resource name prefix, avoiding the chance of name conflict with other users.
```

**Review by rahalan — APPROVED** [review]

**Pull request merged**